src/Microsoft.OData.Core/MediaTypeUtils.cs

```diff
@@ -967,6 +976,27 @@ public override int GetHashCode()
                 int result = this.MediaTypeResolver.GetHashCode() ^ this.PayloadKind.GetHashCode();
                 return this.ContentTypeName != null ? result ^ this.ContentTypeName.GetHashCode() : result;
             }
+
+            /// <summary>
+            /// Multipart/mixed content types have a boundary that varies by request.
+            /// It should not be used as part of the format key.
+            /// </summary>
+            /// <param name="contentTypeName"></param>
+            /// <returns></returns>
+            private static string RemoveBoundary(string contentTypeName)
+            {
+                if (contentTypeName.StartsWith("multipart", StringComparison.OrdinalIgnoreCase))
+                {
+                    // our formatters don't care about parameters for multipart, so just strip them all out for simplicity
+                    int parameter = contentTypeName.IndexOf(';');
+                    if (parameter > 0)

```

Comment:
I assume there will be only 1 ';' in the string, so rather than index and then substring, can we do in 1 pass? may be by using a stringbuilder